src/Spice86.Core/Emulator/VM/Breakpoint/Serializable/EmulatorBreakpointsSerializer.cs

```diff
@@ -0,0 +1,115 @@
+﻿namespace Spice86.Core.Emulator.VM.Breakpoint.Serializable;
+
+using Spice86.Shared.Interfaces;
+using Spice86.Shared.Emulator.VM.Breakpoint.Serializable;
+using Spice86.Core.CLI;
+using Spice86.Shared.Utils;
+
+using System;
+using System.Text.Json;
+using System.Text;
+using System.IO;
+using System.Security.Cryptography;
+
+/// <summary>
+/// Serializes (on startup) and deserialize (on app shutdown) internal debugger breakpoints from disk.
+/// </summary>
+public class EmulatorBreakpointsSerializer {
+    private const string BreakpointsFileNameFormat = "Breakpoints_{0}.json";
+    private readonly Configuration _configuration;
+    private readonly ILoggerService _loggerService;
+    private ISerializableBreakpointsHolder? _serializableBreakpointsHolder = null;
+    private readonly string _programHash;
+
+    public EmulatorBreakpointsSerializer(Configuration configuration, ILoggerService loggerService) {
+        _configuration = configuration;
+        _loggerService = loggerService;
+        _programHash = GetProgramHash(configuration);
+    }
+
+    private string GetProgramHash(Configuration configuration) {
+        if (configuration.ExpectedChecksumValue != null && configuration.ExpectedChecksumValue.Length > 0) {
+            return ConvertUtils.ByteArrayToHexString(configuration.ExpectedChecksumValue);
+        }
+
+        // If no hash is specified,compute it ourselves
+        if (string.IsNullOrWhiteSpace(configuration.Exe) || !File.Exists(configuration.Exe)) {
+            throw new FileNotFoundException(configuration.Exe);
+        }
+        return ConvertUtils.ByteArrayToHexString(SHA256.HashData(File.ReadAllBytes(configuration.Exe)));
+    }
+
+    public void AddSerializableBreakpointsHolder(ISerializableBreakpointsHolder serializableBreakpointsHolder) {
```

Comment:
The method does not "add" a breakpoint holder. Why isn't it initialized from the constructor?

Reply:
Removed - we use ctor dependency injection now.